backend/app/audio.py

```diff
@@ -1,28 +1,28 @@
 """audio processing, etc"""
 from pydub import AudioSegment
+import pandas as pd
+import numpy as np
+import soundfile as sf
 
 
 class Audio:
     silence_threshold = -50.0
-    chunk_size = 10
+    threshold_value = 0.0007  # Tweak the value of threshold to get the aggressive trimming
 
     @staticmethod
-    def _detect_leading_silence(sound: AudioSegment) -> int:
-        trim_ms = 0
-        assert Audio.chunk_size > 0  # to avoid infinite loop
-        while sound[trim_ms:trim_ms + Audio.chunk_size].dBFS \
-                < Audio.silence_threshold and trim_ms < len(sound):
-            trim_ms += Audio.chunk_size
+    def _detect_leading_silence(sound: bytearray, sample_rate: int) -> list:
+        y = pd.Series(sound).apply(np.abs)
+        y_mean = y.rolling(window=int(sample_rate / 20),
+                           min_periods=1,
+                           center=True).max()
 
-        return trim_ms
+        return [True if mean > Audio.threshold_value else False for mean in y_mean]
```

Comment:
The result of a comparison is already a Boolean, eg `x > y` evaluates to either True or False. So this can be simplified to something like:
```
return [mean > Audio.threshold_value for mean in y_mean]
```

Reply:
yup I think this is right